packages/dataviews/src/dataviews-layouts/table/style.scss

```diff
@@ -98,35 +93,16 @@
 				border-top: 1px solid rgba(var(--wp-admin-theme-color--rgb), 0.12);
 			}
 
-			&:hover {
-				background-color: color-mix(in srgb, rgb(var(--wp-admin-theme-color--rgb)) 8%, $white);
-			}
-
 			.dataviews-view-table__actions-column--sticky {
 				background-color: color-mix(in srgb, rgb(var(--wp-admin-theme-color--rgb)) 4%, $white);
 			}
-
-			&:hover .dataviews-view-table__actions-column--sticky {
-				background-color: color-mix(in srgb, rgb(var(--wp-admin-theme-color--rgb)) 8%, $white);
-			}
 		}
 	}
 
-	// Hover styles only apply when bulk actions are available
+	// Show interactive elements on focus
 	&.has-bulk-actions {
 		tr {
-			// Only apply hover background to non-selected rows
-			&:not(.is-selected) {
-				&.is-hovered,
-				&.is-hovered .dataviews-view-table__actions-column--sticky {
-					background-color: #f8f8f8;
-				}
-			}
-
-			&:focus-within,
-			&.is-hovered,
-			&:hover {
-				.components-checkbox-control__input,
```

Comment:
It's safe to remove this since checkboxes are permanently visible now. This should have been removed in https://github.com/WordPress/gutenberg/pull/73248 but was overlooked.